lib/OpenQA/CacheService.pm

```diff
@@ -30,7 +31,28 @@ sub startup {
     $self->defaults(appname => 'openQA Cache Service');
     # Provide help to users early to prevent failing later on
     # misconfigurations
-    return if $ENV{MOJO_HELP};
+    return $self->{_app_return_code} = 0 if $ENV{MOJO_HELP};
+
+    # Stop the service after a critical database error
+    $self->helper(
+        'reply.exception' => sub {
```

Comment:
I believe you could actually just reuse the original helper code and add your extras on top.
```perl
my $code = $self->renderer->get_helper('reply.exception');
$self->helper('reply.exception' => sub {
  my ($c, $error) = @_;
  $c->$code($error);
  $error = $c->stash('exception');
  ...custom code...
});
```
(That's untested, but somethign similar should work)

Reply:
I was thinking the same but honestly didn't know how to access the "original" renderer from there. I'll give it a try.

Reply:
It complains that it can not find `$code` - so this is even too magic for the Perl interpreter.

Reply:
Oh, I've forgotten the first line.

Reply:
I probably shoudl have called it `$sub` instead of `$code` to make it more obvious. It's just the coderef for the helper sub.